lib/formatters/base.ts

```diff
@@ -24,35 +24,37 @@
 
 import * as exec from '../exec';
 
-export class BaseFormatter {
-    constructor(formatterInfo) {
+import {FormatOptions, FormatterInfo} from './base.interfaces';
+
+// TODO(supergrecko): initial type for ../exec.js exec method
+export type ExecResult = {
```

Comment:
You can use https://github.com/compiler-explorer/compiler-explorer/blob/main/types/execution/execution.interfaces.ts#L3